execution_engine/src/storage/trie/mod.rs

```diff
@@ -521,6 +524,20 @@ impl<K, V> Trie<K, V> {
             _ => None,
         }
     }
+
+    /// Returns the hash of this Trie.
+    pub fn hash(&self) -> Result<Digest, TrieHashingError>
+    where
+        Self: ToBytes,
+    {
+        let bytes = self.to_bytes()?;
+        if bytes.len() <= ChunkWithProof::CHUNK_SIZE_BYTES {
+            Ok(Digest::hash(&bytes))
+        } else {
+            let chunk_with_proof = ChunkWithProof::new(&bytes, 0)?;
```

Comment:
Actually, do we have a method that computes the hash without copying a chunk? Would be better to avoid the unnecessary copying.

Reply:
I think we don't, but I agree it would be a good idea.

Reply:
...Or maybe we do :thinking:

Reply:
(Either way, something that can also be added later, in a separate PR.)